extensions/panache/panache-mock/src/main/java/io/quarkus/panache/mock/PanacheMock.java

```diff
@@ -40,6 +41,13 @@ public static void mock(Class<?>... classes) {
         IsMockEnabled = !mocks.isEmpty();
     }
 
+    public static void mock(MockSettings settings, Class<?>... classes) {
+        for (Class<?> klass : classes) {
+            mocks.computeIfAbsent(klass, v -> Mockito.mock(klass, settings));
+        }
+        IsMockEnabled = !mocks.isEmpty();
```

Comment:
you have a race condition here; maybe it's not an issues as I don't expect many will invoke this method concurrecntly, but perhaps you'd still prefer to use a safer locking pattern?

I don't think this needs to be "high throughput" or high performance, so just synchronize it all?

Reply:
OK